### ISSUE TITLE
Don't prompt for yes/no answer if input is not a TTY

### ISSUE THREAD

**Issue by kevinburkeotto** [gruntwork-io/terragrunt#317]
We call Terragrunt from inside of a CI environment. At one point, Terragrunt stops and asks:

```
Remote S3 state bucket ... does not exist. Would you like Terragrunt to create it? (y/n) 
```

However, there's no way to answer that question, since it's running in a CI environment. Terragrunt can determine this at runtime by checking whether stdin is a tty; if not, Terragrunt should exit with an error.

(It's currently hanging indefinitely)

**Comment by josh-padnick** [comment]
@kevinburkeotto I would argue that, for most teams, this particular prompt should not be handled by CI. An operator should manually create the S3 Bucket where Terraform state is going to live, and CI should be running commands that _update_ that state, but if CI prompts you to create it, that may well be an indication that your code is attempting to create an S3 Bucket in the wrong account, or for the wrong directory.

In other words, in most cases, the fact that CI hangs on this winds up functioning as a de facto guardrail. That being said, it's not a first-class error and requires some digging. 

Do you have a use case where it would make sense for CI to _create_ the S3 Bucket? That would imply you're dynamically creating and applying lots of new Terraform configurations on the fly.

**Comment by brikis98** [comment]
FWIW, there is a `--terragrunt-non-interactive` flag that assumes "yes" for all prompts: https://github.com/gruntwork-io/terragrunt#cli-options

**Comment by kevinburkeotto** [comment]
> the fact that CI hangs on this winds up functioning as a de facto guardrail.

I agree, but I think the behavior here should be to exit with an error instead of to hang indefinitely. On some CI environments, you have a finite number of workers, and hanging means that no other workers can continue.
